pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,365 @@
+	// Create creates a container in the sandbox, and returns the ID
+	// of the created container.
+	Create(config *ContainerConfig, sandboxConfig *PodSandboxConfig, sandboxID PodSandboxID) (RawContainerID, error)
+	// Start starts a created container.
+	Start(id RawContainerID) error
+	// Stop stops a running container with a grace period (i.e., timeout).
+	Stop(id RawContainerID, timeout int) error
+	// Remove removes the container.
+	Remove(id RawContainerID) error
+	// List lists the existing containers that match the ContainerFilter.
+	// The returned list should only include containers previously created
+	// by this ContainerManager.
+	List(filter ContainerFilter) ([]Container, error)
+	// Status returns the status of the container.
+	Status(id RawContainerID) (RawContainerStatus, error)
+	// Exec executes a command in the container.
+	Exec(id RawContainerID, cmd []string, streamOpts StreamOptions) error
+}
+
+type ContainerConfig struct {
+	// Name of the container. The string should conform to [a-zA-Z0-9_-]+.
+	Name string
+	// Image to use.
+	Image ImageSpec
+	// Command to execute (i.e., entrypoint for docker)
+	Command []string
+	// Args for the Command (i.e., command for docker)
+	Args []string
+	// Current working directory of the command.
+	WorkingDir string
+	// List of environment variable to set in the container
+	Env []KeyValue
+	// Mounts specifies mounts for the container
+	Mounts []Mount
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+	// Privileged runs the container in the privileged mode.
+	Privileged bool
+	// ReadOnlyRootFS sets the root filesystem of the container to be
+	// read-only.
+	ReadOnlyRootFS bool
+	// Path relative to PodSandboxConfig.LogDirectory for container to store
+	// the log on the host.
+	LogPath string
+
+	// Variables for interactive containers, these have very specialized
+	// use-cases (e.g. debugging).
+	// TODO: Determine if we need to continue supporting these fields that are
+	// part of Kubernetes's Container Spec.
+	STDIN     bool
+	STDINONCE bool
+	TTY       bool
+
+	// Linux contains configuration specific to Linux containers.
+	Linux *LinuxContainerConfig
+}
+
+// RawContainerStatus represents the status of a container.
+type RawContainerStatus struct {
+	// ID of the container.
+	ID ContainerID
+	// Name of the container.
+	Name string
+	// Status of the container.
+	State ContainerState
+	// Creation time of the container.
+	CreatedAt unversioned.Time
+	// Start time of the container.
+	StartedAt unversioned.Time
+	// Finish time of the container.
+	FinishedAt unversioned.Time
+	// Exit code of the container.
+	ExitCode int
+	// Name of the image, this also includes the tag of the image,
+	// the expected form is "NAME:TAG".
+	Image string
```

Comment:
The type is only a string now, but the comment did mention "tag".
I'm not familiar with other image formats e.g. ACI. Is string not enough for ACI name? Is there tag in ACI image format? If there is no tag concept in ACI image format, is there any equivalent concept? :)
Maybe we can update the comment, with some "e.g.".